Test_Cases.py

```diff
@@ -1,169 +1,216 @@
+
+        Note
+        --------------------
+        -Yahoo security will randomly ask to "prove you are not a robot".
+         Which, at times, can be responsible for the tests in Error Message
+         Passwords class to error. There is not an easy fix for this.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(5))
+
+        self.assertEqual(yahoo_page.short_password_error_message().
+                         get_attribute('data-error'), "WEAK_PASSWORD")
+        self.assertEqual(yahoo_page.short_password_error_message().text,
+                         "- Your password is too easy to guess.")
+
+    def test_moderate_password_error_message(self):
+        """
+        Confirm that when the password 9 or 10 characters in length, the
+        error data status is "ALMOST_THERE". Confirm the resulting error
+        text is "- Almost there".
+
+        acceptance criteria
+        --------------------
+        -When a password that is between 9 and 10 characters is entered,
+         the data error status and the error text exhibit's this is a
+         moderate length for the password.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(9))
+        time.sleep(0.75)  # Please look at the branch Explicit_Wait_ErrorMessagePasswords if you want to see why I used this implicit wait. This keeps my code clean and readable. No E.C.s exist that can wait for this dynamic element correctly for this specific test. So I created my own E.C; which you can see being used in the given branch. This E.C. is also in Tools in this repo.
+
+        self.assertEqual(yahoo_page.short_password_error_message().
+                         get_attribute('data-error'), "ALMOST_THERE")
+        self.assertEqual(yahoo_page.moderate_password_error_message().text,
+                         "- Almost there.")
+
+    def test_long_password_error_message(self):
+        """
+        Confirm that when the password is 11 characters, the error data
+        status is "STRONG_PASSWORD". Confirm the resulting error text is
+        "- you did it!".
+
+        acceptance criteria
+        --------------------
+        -When a password that is 11 characters is entered, the data
+         error and error text either does not exist or it exhibits this
+         is a password of sufficient length.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(11))
+        time.sleep(1)  # The Explicit_Wait_ErrorMessagePasswords branch has some really cool stuff in it. But if you look at it you will see why I decided to use this implicit wait instead
+
+        self.assertEqual(yahoo_page.long_password_error_message().
+                         get_attribute('data-error'), "STRONG_PASSWORD")
+        self.assertTrue(yahoo_page.long_password_error_message().text == "" or
```

Comment:
You should have only one assert statement per test, regardless of the corner cases you'd like to test after the first assert. If you really need to validate all those clauses in this test, then use only `assertTrue` that validates that all your conditions have been met.

That said, a good practice for automated testing is to avoid tests that assert too many things... each test should be atomic to avoid failures to a whole test because one of the conditions in the assertion failed.